### PR TITLE
Issue #3949: NoWhitespaceBefore: add support for method reference operator

### PR DESCRIPTION
**Pull request by Kietzmann**
This PR related to issue #3949 
What was done:
1) added new tokens into config
2) added new test class into /nowhitespacebefore package
3) added new test method in NoWhitespaceBeforeCheckTest.java to test correct behavior.

### CODE REVIEW

File: config/checkstyle_checks.xml [hunk 1]
```diff
@@ -474,6 +474,7 @@
     <module name="NoWhitespaceBefore"/>
     <module name="NoWhitespaceBefore">
       <property name="tokens" value="DOT"/>
+      <property name="tokens" value="METHOD_REF"/>
```
Comment (rnveach) [inline config/checkstyle_checks.xml]: @romani Please confirm you are ok with this token here.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      }
+    }
+  }
+
+  public static class Nested2<V> {
+  }
+
+  public <V> void methodName(V value) {
+    Supplier<?> t = Nested2<V> ::new; //19:31 violation
+    Supplier<SomeClass.Nested<V>> passes = SomeClass.Nested ::new; //20:60 violation
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java]: Just put violation, not the line/column.
If input is updated, no one will remember to change these positionings.

Comment (Kietzmann) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java]: Done

File: config/checkstyle_checks.xml (hunk 1 above)
Comment (romani) [inline config/checkstyle_checks.xml]: confirmed

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com . puppycrawl
+    .tools.
+    checkstyle.checks.whitespace.nowhitespacebefore;
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java]: please simplify package - make it single line.
in this input you should focus on NoWhitespaceBeforeMethodRef

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    }
+  }
+
+  public static class Nested2<V> {
+  }
+
+  public <V> void methodName(V value) {
+    Supplier<?> t = Nested2<V> ::new; //violation
+    Supplier<SomeClass.Nested<V>> passes = SomeClass.Nested ::new; //violation
+    Supplier<SomeClass.Nested<V>> fails = SomeClass.Nested<V>::new; //no violation
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java]: please remove "//no violation"

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java (hunk 3 above)
Comment (Kietzmann) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java (hunk 4 above)
Comment (Kietzmann) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java]: Done

**Pull request merged**